src/main/java/yasp/Parse.java

```diff
@@ -459,6 +459,19 @@ public void onTickStart(Context ctx, boolean synthetic) {
             }
         }
     }
+    
+    @OnWardCountered
+    public void onWardCountered(Context ctx, Entity e, String killer_hero_name) {
+        Entry wardEntry = buildWardEntry(ctx, e);
+        wardEntry.attackername = killer_hero_name;
+        output(wardEntry);
+    }
+    
+    @OnWardExpired
+    @OnWardPlaced
+    public void onWardExistenceChanged(Context ctx, Entity e) {
```

Comment:
can you unify this method with the previous one and detect from context which case it is?